test/distributed/test_c10d.py

```diff
@@ -2406,11 +2407,11 @@ def _test_gloo_backend(
 
     @requires_gloo()
     def test_gloo_backend_cpu_module(self):
-        self._test_gloo_backend([torch.device("cpu")], [])
+        self._test_gloo_backend([torch.device("cpu")], None)
 
     @requires_gloo()
     def test_gloo_backend_cpu_module_grad_is_view(self):
-        self._test_gloo_backend([torch.device("cpu")], [], gradient_as_bucket_view=True)
+        self._test_gloo_backend([torch.device("cpu")], None, gradient_as_bucket_view=True)
```

Comment:
any reason for changing empty list to None? For BC, we need to confirm that empty list still works, no?

Reply:
It seems the review suggestions have gone back and forth, from 1) keeping the empty list option for BC, 2) not keeping for the consistency with multi-device module, to 3) now keeping the empty list option for BC, but not documenting this option.

I have confirmed the empty list can still work, by keeping only a new unit test `test_nccl_backend_single_device_module_empty_device_ids`. Reverting these changes to the existing unit tests can still work, but since we don't want to mention this in the documentation, prefer changing the empty list to the advocated `None` for the most tests. After all, unit tests can also be viewed as a kind of documentation in some degree.